### PR TITLE
feat: migrate to Ruff for code linting and formatting

### PR DESCRIPTION
**Pull request by jorgepiloto**
This pull-request drops support for black, isort, and flake in favor of [Ruff](https://docs.astral.sh/ruff), which contains all previous tools and more!

This linter and formatter is really fast for big projects. Its rules are able to identify common issues and format files in just a few milliseconds.

This pull-request is related with #1722.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-        doc/source/|
-        src/ansys/dpf/core/operators/|
-        examples_notebooks/|
-        src/ansys/dpf/core/__init__.py|
-        doc/styles/|
-        src/ansys/dpf/gate/|
-        src/ansys/dpf/gatebin/|
-        src/ansys/grpc/dpf/|
-    )
+exclude: >
```
Comment (jorgepiloto) [inline .pre-commit-config.yaml:1]: Without fixing this, pre-commit was ignoring all files.

**Comment by jorgepiloto** [comment]
As discussed in private, we will enable step by step the different rules for better fixing any issues. Despite the rules being deactivated, Ruff formatter still provides all current capabilities. Just see the rules as "extras" or "plugins".

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,31 +1,28 @@
...
+    - id: ruff
+      args: [ --fix ]
+    - id: ruff-format
 
 - repo: https://github.com/codespell-project/codespell
   rev: v2.3.0
   hooks:
   - id: codespell
-    args: ["--ignore-words=ignore_words.txt"]
+    args: ["--ignore-words=./doc/styles/Vocab/ANSYS/accept.txt"]
```
Comment (jorgepiloto) [inline .pre-commit-config.yaml:25]: Unifying all ignore/accepted words in the "accept.txt" file

File: doc/styles/Vocab/ANSYS/accept.txt
Comment (jorgepiloto) [inline doc/styles/Vocab/ANSYS/accept.txt:1]: I just sorted and uniq this file

**Review by PProfizi — APPROVED** [review]
Checked the main changes, scanned the rest quickly, LGTM

**Pull request merged**